backend/routers/spiritual.py

```
@@ -544,28 +545,38 @@ async def link_anonymous_chart_to_user(request: Request):
     user_email = extract_user_email_from_token(request)
     if not user_email:
         raise HTTPException(status_code=401, detail="Authentication required")
     
     try:
-        # Connect to database to get user sessions and progress
-        from db import get_db
+        # Convert user_id to integer for validation
+        try:
+            user_id_int = int(user_id)
+        except (ValueError, TypeError):
+            raise HTTPException(status_code=400, detail="Invalid user ID format")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)